src/main/java/com/hyperwallet/clientsdk/util/HyperwalletEncryption.java

```diff
@@ -0,0 +1,182 @@
+                new Payload(jwsObject));
+
+        jweObject.encrypt(new RSAEncrypter((RSAKey)hyperwalletPublicKey));
+
+        return jweObject.serialize();
+    }
+
+    public String decrypt(String body) throws ParseException, IOException, JOSEException {
+
+        JWK privateKeyToDecrypt = getKeyByAlgorithm(loadKeySet(clientPrivateKeySetLocation), encryptionAlgorithm);
+        RSAPublicKey publicKeyToSign = ((RSAKey)getKeyByAlgorithm(loadKeySet(hyperwalletKeySetLocation), signAlgorithm))
+                .toRSAPublicKey();
+
+        JWEObject jweObject = JWEObject.parse(body);
+        jweObject.decrypt(new RSADecrypter((RSAKey)privateKeyToDecrypt));
+        JWSObject jwsObject = jweObject.getPayload().toJWSObject();
+        verifySignatureExpirationDate(jwsObject.getHeader().getCustomParam(EXPIRATION));
+        boolean verifyStatus = jwsObject.verify(new RSASSAVerifier(publicKeyToSign,
+                new HashSet<>(Collections.singletonList(EXPIRATION))));
+        if(!verifyStatus) {
+            throw new HyperwalletException("JWS signature is incorrect");
+        }
+        return jwsObject.getPayload().toString();
+    }
+
+    public void verifySignatureExpirationDate(Object signatureExpirationDate) {
+        if (signatureExpirationDate == null) {
+            throw new HyperwalletException("exp JWS header param was null");
+        }
+        if (!(signatureExpirationDate instanceof Long)) {
+            throw new HyperwalletException("exp JWS header must be of type Long");
+        }
+        long expirationTimeSeconds = (long)signatureExpirationDate;
+        if (new Date().toInstant().getEpochSecond() > expirationTimeSeconds) {
+            throw new HyperwalletException("Response message signature(JWS) has expired");
+        }
+    }
+
+    public JWEAlgorithm getEncryptionAlgorithm() {
+        return encryptionAlgorithm;
+    }
+
+    public void setEncryptionAlgorithm(JWEAlgorithm encryptionAlgorithm) {
+        this.encryptionAlgorithm = encryptionAlgorithm;
+    }
+
+    public HyperwalletEncryption encryptionAlgorithm(JWEAlgorithm encryptionAlgorithm) {
+        this.encryptionAlgorithm = encryptionAlgorithm;
+        return this;
+    }
+
+    public JWSAlgorithm getSignAlgorithm() {
+        return signAlgorithm;
+    }
+
+    public void setSignAlgorithm(JWSAlgorithm signAlgorithm) {
+        this.signAlgorithm = signAlgorithm;
+    }
+
+    public HyperwalletEncryption signAlgorithm(JWSAlgorithm signAlgorithm) {
+        this.signAlgorithm = signAlgorithm;
+        return this;
+    }
+
+    public EncryptionMethod getEncryptionMethod() {
+        return encryptionMethod;
+    }
+
+    public void setEncryptionMethod(EncryptionMethod encryptionMethod) {
+        this.encryptionMethod = encryptionMethod;
+    }
+
+    public HyperwalletEncryption encryptionMethod(EncryptionMethod encryptionMethod) {
+        this.encryptionMethod = encryptionMethod;
+        return this;
```

Comment:
why do we need to do like this? we don't need to expose old builder pattern implementation, we can always do like this `GenericBuilder`